.github/workflows/community-review.sh

```diff
@@ -0,0 +1,470 @@
+    else
+      all_reviews="$(get_all_reviews_for_pr "$token" "$pr_number")" 
+    fi
+    # leave only the latest reviews per reviewer in a comma separated form
+    pr_reviewers="$(jq  '. | sort_by([.author.login, .createdAt]) | reverse | unique_by(.author.login) | .[] | [.author.login, .state, .createdAt] | join(",")' <<< "$all_reviews")"
+    
+    printf "Reviews %s Reviewers %s\n" "$(JSONArrayLength "$all_reviews")" "$(wc -l <<< "$pr_reviewers" | xargs)"
+    
+    process_pr_reviews "$token" "$pr_number" "$pr_reviewers" || exit
+    ((counter++))
+  done <<< "$prNumbersAndPaging" || exit
+}
+
+# =============================================================================
+# Process pr reviews for a pr
+# The pr reviews a line for each review, with the user, creation time and review state
+# This function processes the pr reviews for the pr to obtain
+#  community approves
+#  request for changes
+#  committer approves
+#  community reviews
+#
+# If there are 2 or more community approves, no request for changes and no committer approves we set
+# label 'community-reviewed-LGTM' on the PR. If not and there has been a community review, label
+# 'community-reviewed' is set.
+# If one of the labels is set the other is unset.
+#
+# Arguments:
+#   $1 - GitHub API token for authentication
+#   $2 - PR number
+#   $3 - PR reviews
+# =============================================================================
+process_pr_reviews() {
+  local token="${1?missing token}"
+  local pr_number="${2?missing pr number}"
+  local pr_reviews="${3?missing pr reviews}"
+
+  local communityApproves=0
+  local requestForChanges=0
+  local committerApproves=0
+  local communityReviews=0
+  local push_permission
+  # replace spaces with new lines so the loop will work
+  pr_reviews=$(echo "$pr_reviews" | tr ' ' '\n')
+  # remove unnecessary double quotes
+  pr_reviews="${pr_reviews//\"/}"
+
+  while IFS=, read -r user state time
+  do
+      printf "%-15s | %-20s | %-20s - checking user permissions..." "$user" "$state" "$time"
+      push_permission=$(call_github_get_user_push_permission "$token" "$user") || exit
+      printf "%s\n" "$push_permission"
+
+      #see if the user has read role
+
+      if [ "$push_permission" = 'true' ]; then
+          if [ "$state" = "APPROVED" ]; then
+             ((++committerApproves))
+          fi
+     else
+          ((++communityReviews))
+          if [ "$state" = "APPROVED" ]; then
+             ((++communityApproves))
+          fi
+     fi
+
+     if [ "$state" = "CHANGES_REQUESTED" ]; then
+        ((++requestForChanges))
+     fi
+  done <<< "$pr_reviews"
+  echo "communityApproves $communityApproves requestForChanges $requestForChanges committerApproves $committerApproves communityReviews $communityReviews"
+
+  local label_to_post=
+  local label_to_delete=
+  if [[ $communityApproves -ge  "2" && $requestForChanges = 0 && $committerApproves = 0 ]]; then
```

Comment:
I'd suggest to not mix string and arithmetic comparison, although here it would produce the desired result, if this gets modified it won't necessarily be true after that. For checking numbers, IMO rule of thumb in bash is to use arithmetic comparison operator, hence this condition should be:
```bash
[[ $communityApproves -ge  2 && $requestForChanges -eq 0 && $committerApproves -eq 0 ]]
```